static-site/components/list-resources/index.tsx

```diff
@@ -142,14 +147,27 @@ function ListResourcesContent({
   defaultGroupLinks = false,
   groupDisplayNames,
   tileData,
+  resourceName,
 }: ListResourcesProps & {
   elWidth: number;
 }): React.ReactElement {
+  let resourceListingCallback: () => Promise<ResourceListingInfo>;
+  switch(resourceName) {
+    case "pathogens":
+      resourceListingCallback = pathogenResourceListingCallback;
+      break;
+    case "staging":
+      resourceListingCallback = stagingResourceListingCallback;
+      break;
+    default:
+      throw new InternalError(`${resourceName} is not a valid resource name`);
+  }
```

Comment:
Suggestion: pass callback function directly (ac4e552995d4b28dcd6ed7858626cff4b6e24993)

Reply:
Again, memorializing a decision made in a conversation: I will move the callback usage in the direction Victor suggests here; I had previously tried this approach and failed, not understanding that the callback file needs to be a React Client Component.